### DISCUSSION TITLE
Potentials > limitless

### DISCUSSION THREAD

**Discussion by magick93** [cueblox/blox#106] [General]
This is a really interesting project! Thanks for building and sharing this.

I have a few questions.

Would it be possible to pull in cue schema such as this https://github.com/adieu/cue-ansible ?

Is it possible to have references from schema to another? And furthermore, to do something equivalent to sql foreign keys? Eg, using the `#Person` and `#Article` examples - can I ensure there is a constraint that a new article must reference an existing person?

**Comment by rawkode** [comment]
I'm sorry again @magick93 for not seeing these discussions, I'll fix my notifications.

We do support foreign key constraints through `@relationship` annotation and passing the `--referential-integrity` (or `-i`) flag to `blox build`

You can see this in action in our `./dogfood` directories and my own project: https://github.com/rawkode-academy/rawkode-academy
